ENIGMAsystem/SHELL/Audio_Systems/DirectSound/DSsystem.cpp

```diff
@@ -82,10 +82,9 @@ int audiosystem_initialize() {
   DSBUFFERDESC bufferDesc;
 
   // Initialize the direct sound interface pointer for the default sound device.
-  result = DirectSoundCreate8(NULL, &dsound, NULL);
-  if (FAILED(result)) {
+  // Allow the user to retry until a device is created.
+  while (FAILED(result = DirectSoundCreate8(NULL, &dsound, NULL))) {
```

Comment:
Your loop condition doesn't have a check for permanent errors vs transient. Such a check may not exist. Consider adding a sleep, and consider adding a maximum retry count. It's probably better for the game to scream a lot, then continue with no sound, rather than hanging forever, printing "Failed to create a DirectSound8 object."

Reply:
Well there is only one error here that can be retried, if anything fails after this is successful, it should still be treated as fatal as it still is. Also those seem like good ideas.

And your view that it should scream a lot seems to be exactly the opposite of where YoYoGames went with GMS. As I said in #1770 it seems GMS now handles this as a silent fail, warns about it, and loads a dummy audio engine. It doesn't really hang forever like it is now though, the user has both the "Abort" and "Ignore" (i'm using retry as a synonym here; there are only 2 buttons on this dialog now because of tkg) options in this scenario.